ftdetect/ninja.vim

```diff
@@ -0,0 +1,2 @@
+autocmd BufRead,BufNewFile *.ninja setfiletype ninja
+
```

Comment:
I think Neovim already has Ninja filetype detection. At least I'm guessing that from grepping.
```
❯ rg ninja
unicode/emoji-data.txt
377:1F977..1F978  ; Emoji                # E13.0  [2] (🥷..🥸)    ninja..disguised face
646:1F977..1F978  ; Emoji_Presentation   # E13.0  [2] (🥷..🥸)    ninja..disguised face
735:1F977         ; Emoji_Modifier_Base  # E13.0  [1] (🥷)       ninja
1216:1F977..1F978  ; Extended_Pictographic# E13.0  [2] (🥷..🥸)    ninja..disguised face

src/nvim/testdir/test_filetype.vim
352:    \ 'ninja': ['file.ninja'],

CMakeLists.txt
616:  # When running tests from 'ninja' we need to use the

ci/build.ps1
64:  $mingwPackages = @('ninja', 'cmake', 'diffutils').ForEach({

no_build_type/CMakeCache.txt
144://Program used to build from build.ninja files.
145:CMAKE_MAKE_PROGRAM:FILEPATH=/usr/bin/ninja

ci/common/build.sh
13:  ninja "$@"

no_build_type/runtime/cmake_install.cmake
41042:  file(INSTALL DESTINATION "${CMAKE_INSTALL_PREFIX}/share/nvim/runtime/syntax" TYPE FILE PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ FILES "/home/stephan/projects/neovim/runtime/syntax/ninja.vim")

no_build_type/build.ninja
21:ninja_required_version = 1.5
35:include CMakeFiles/rules.ninja
17501:build build.ninja: RERUN_CMAKE | ../CMakeLists.txt ../cmake/FindIconv.cmake ../cmake/FindLIBVTERM.cmake ../cmake/FindLibIntl.cmake ../cmake/FindLibLUV.cmake ../cmake/FindLibTermkey.cmake ../cmake/FindLibUV.cmake ../cmake/FindLuaJit.cmake ../cmake/FindMsgpack.cmake ../cmake/FindUNIBILIUM.cmake ../cmake/GetCompileFlags.cmake ../cmake/InstallHelpers.cmake ../cmake/LibFindMacros.cmake ../cmake/LuaHelpers.cmake ../cmake/PreventInTreeBuilds.cmake ../config/CMakeLists.txt ../config/config.h.in ../config/pathdef.c.in ../config/versiondef.h.in ../runtime/CMakeLists.txt ../src/nvim/CMakeLists.txt ../src/nvim/po/CMakeLists.txt ../test/config/paths.lua.in ../test/functional/fixtures/CMakeLists.txt ../test/includes/CMakeLists.txt /snap/cmake/549/share/cmake-3.18/Modules/CMakeCCompiler.cmake.in /snap/cmake/549/share/cmake-3.18/Modules/CMakeCCompilerABI.c /snap/cmake/549/share/cmake-3.18/Modules/CMakeCInformation.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeCheckCompilerFlagCommonPatterns.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeCommonLanguageInclude.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeCompilerIdDetection.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeConfigurableFile.in /snap/cmake/549/share/cmake-3.18/Modules/CMakeDetermineCCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeDetermineCompileFeatures.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeDetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeDetermineCompilerABI.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeDetermineCompilerId.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeDetermineSystem.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeFindBinUtils.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeGenericSystem.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeInitializeConfigs.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeLanguageInformation.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeNinjaFindMake.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeParseImplicitIncludeInfo.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeParseImplicitLinkInfo.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeSystem.cmake.in /snap/cmake/549/share/cmake-3.18/Modules/CMakeSystemSpecificInformation.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeSystemSpecificInitialize.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeTestCCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/CMakeTestCompilerCommon.cmake /snap/cmake/549/share/cmake-3.18/Modules/CPack.cmake /snap/cmake/549/share/cmake-3.18/Modules/CPackComponent.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckCCompilerFlag.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckCSourceCompiles.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckCSourceRuns.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckFunctionExists.c /snap/cmake/549/share/cmake-3.18/Modules/CheckFunctionExists.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckIPOSupported.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckIPOSupported/CMakeLists-C.txt.in /snap/cmake/549/share/cmake-3.18/Modules/CheckIPOSupported/foo.c /snap/cmake/549/share/cmake-3.18/Modules/CheckIPOSupported/main.c /snap/cmake/549/share/cmake-3.18/Modules/CheckIncludeFile.c.in /snap/cmake/549/share/cmake-3.18/Modules/CheckIncludeFile.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckIncludeFileCXX.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckIncludeFiles.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckLibraryExists.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckSymbolExists.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckTypeSize.c.in /snap/cmake/549/share/cmake-3.18/Modules/CheckTypeSize.cmake /snap/cmake/549/share/cmake-3.18/Modules/CheckVariableExists.c /snap/cmake/549/share/cmake-3.18/Modules/CheckVariableExists.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/ADSP-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/ARMCC-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/ARMClang-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/AppleClang-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Borland-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Bruce-C-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/CMakeCommonCompilerMacros.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Clang-C.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Clang-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Clang-DetermineCompilerInternal.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Clang-FindBinUtils.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Clang.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Compaq-C-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Cray-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Embarcadero-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Fujitsu-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/GHS-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/GNU-C-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/GNU.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/HP-C-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/IAR-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/IBMCPP-C-DetermineVersionInternal.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Intel-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/MSVC-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/NVIDIA-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/OpenWatcom-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/PGI-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/PathScale-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/SCO-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/SDCC-C-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/SunPro-C-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/TI-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/TinyCC-C-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/VisualAge-C-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/Watcom-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/XL-C-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/XLClang-C-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/Compiler/zOS-C-DetermineCompiler.cmake /snap/cmake/549/share/cmake-3.18/Modules/FindGettext.cmake /snap/cmake/549/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake /snap/cmake/549/share/cmake-3.18/Modules/FindPackageMessage.cmake /snap/cmake/549/share/cmake-3.18/Modules/FindPkgConfig.cmake /snap/cmake/549/share/cmake-3.18/Modules/FindThreads.cmake /snap/cmake/549/share/cmake-3.18/Modules/GNUInstallDirs.cmake /snap/cmake/549/share/cmake-3.18/Modules/Internal/CMakeCheckCompilerFlag.cmake /snap/cmake/549/share/cmake-3.18/Modules/Internal/FeatureTesting.cmake /snap/cmake/549/share/cmake-3.18/Modules/Platform/Linux-Clang-C.cmake /snap/cmake/549/share/cmake-3.18/Modules/Platform/Linux-GNU-C.cmake /snap/cmake/549/share/cmake-3.18/Modules/Platform/Linux-GNU.cmake /snap/cmake/549/share/cmake-3.18/Modules/Platform/Linux.cmake /snap/cmake/549/share/cmake-3.18/Modules/Platform/UnixPaths.cmake /snap/cmake/549/share/cmake-3.18/Templates/CPackConfig.cmake.in CMakeCache.txt CMakeFiles/3.18.2/CMakeCCompiler.cmake CMakeFiles/3.18.2/CMakeSystem.cmake CMakeFiles/CheckIncludeFiles/HAVE_ENDIAN_H.c CMakeFiles/CheckIncludeFiles/HAVE_LANGINFO_H.c CMakeFiles/CheckIncludeFiles/HAVE_LOCALE_H.c CMakeFiles/CheckIncludeFiles/HAVE_PWD_H.c CMakeFiles/CheckIncludeFiles/HAVE_STRINGS_H.c CMakeFiles/CheckIncludeFiles/HAVE_SYS_ENDIAN_H.c CMakeFiles/CheckIncludeFiles/HAVE_SYS_UIO_H.c CMakeFiles/CheckIncludeFiles/HAVE_SYS_UTSNAME_H.c CMakeFiles/CheckIncludeFiles/HAVE_SYS_WAIT_H.c CMakeFiles/CheckIncludeFiles/HAVE_TERMIOS_H.c CMakeFiles/CheckTypeSize/SIZEOF_INT.c CMakeFiles/CheckTypeSize/SIZEOF_INTMAX_T.c CMakeFiles/CheckTypeSize/SIZEOF_LONG.c CMakeFiles/CheckTypeSize/SIZEOF_LONG_LONG.c CMakeFiles/CheckTypeSize/SIZEOF_SIZE_T.c CMakeFiles/CheckTypeSize/SIZEOF_VOID_PTR.c test/config/paths.lua.gen

runtime/syntax/ninja.vim
1:" ninja build file syntax.
2:" Language: ninja build file as described at
3:"           http://ninja-build.org/manual.html
11:" ninja lexer and parser are at
12:" https://github.com/ninja-build/ninja/blob/master/src/lexer.in.cc
13:" https://github.com/ninja-build/ninja/blob/master/src/manifest_parser.cc
27:syn match ninjaComment /\(\$\n\)\@<!\_^\s*#.*$/  contains=@Spell
32:syn match ninjaKeyword "^build\>"
33:syn match ninjaKeyword "^rule\>"
34:syn match ninjaKeyword "^pool\>"
35:syn match ninjaKeyword "^default\>"
36:syn match ninjaKeyword "^include\>"
37:syn match ninjaKeyword "^subninja\>"
44:syn region ninjaRule start="^rule" end="^\ze\S" contains=TOP transparent
45:syn keyword ninjaRuleCommand contained containedin=ninjaRule command
49:syn region ninjaPool start="^pool" end="^\ze\S" contains=TOP transparent
50:syn keyword ninjaPoolCommand contained containedin=ninjaPool  depth
62:syn match   ninjaDollar "\$\$"
63:syn match   ninjaWrapLineOperator "\$$"
64:syn match   ninjaSimpleVar "\$[a-zA-Z0-9_-]\+"
65:syn match   ninjaVar       "\${[a-zA-Z0-9_.-]\+}"
72:syn match ninjaOperator "\(=\|:\||\|||\)\ze\s"
74:hi def link ninjaComment Comment
75:hi def link ninjaKeyword Keyword
76:hi def link ninjaRuleCommand Statement
77:hi def link ninjaPoolCommand Statement
78:hi def link ninjaDollar ninjaOperator
79:hi def link ninjaWrapLineOperator ninjaOperator
80:hi def link ninjaOperator Operator
81:hi def link ninjaSimpleVar ninjaVar
82:hi def link ninjaVar Identifier
84:let b:current_syntax = "ninja"

runtime/lua/vim/filetype.lua
388:  ninja = "ninja",

runtime/makemenu.vim
410:SynMenu NO.Ninja:ninja

runtime/synmenu.vim
387:an 50.80.160 &Syntax.NO.Ninja :cal SetSyn("ninja")<CR>

runtime/filetype.vim
1170:au BufNewFile,BufRead *.ninja                      setf ninja

CONTRIBUTING.md
36:- Install `ninja` for faster builds of Nvim.
38:  sudo apt-get install ninja-build
40:  make  # Nvim build system uses ninja automatically, if available.

snap/snapcraft.yaml
74:            - ninja-build

Makefile
41:CMAKE_GENERATOR ?= $(shell (command -v ninja > /dev/null 2>&1 && echo "Ninja") || \
51:      BUILD_TOOL = ninja

test/functional/lua/filetype_spec.lua
410:      ['ninja'] = {'file.ninja'}
``

Reply:
Ops.